src/gar.lua

```diff
@@ -187,21 +187,19 @@ function gar.leaveNetwork(from, currentTimestamp, msgId)
 
 	-- if the slash happens to be 100% we do not need to vault anything
 	if minimumStakedTokens > 0 then
-		gateway.vaults[from] = {
-			balance = minimumStakedTokens,
-			startTimestamp = currentTimestamp,
-			endTimestamp = gatewayEndTimestamp,
-		}
+		gar.createGatewayWithdrawVault(gateway, from, minimumStakedTokens, currentTimestamp, gatewayEndTimestamp)
 
 		-- if there is more than the minimum staked tokens, we need to vault the rest but on shorter term
 		local remainingStake = gateway.operatorStake - gar.getSettings().operators.minStake
 
 		if remainingStake > 0 then
-			gateway.vaults[msgId] = {
-				balance = remainingStake,
-				startTimestamp = currentTimestamp,
-				endTimestamp = gatewayStakeWithdrawTimestamp,
-			}
+			gar.createGatewayWithdrawVault(
```

Comment:
I'd argue that we shouldn't let the endTimestamp be a variable here. We should centralize the logic for computing it in the function.

Reply:
[7ff44c4](https://github.com/ar-io/ar-io-network-process/pull/269/commits/7ff44c458ca916a481396b573ca7942ffb8cc3d8)

Reply:
Pretty nice! My inner FYEO is thinking some more assertions for input validation in those new gar functions might be good. But looking good so far.